### PR TITLE
feat: updated deployment scripts

### PR DESCRIPTION
**Pull request by sogipec**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -15,5 +15,6 @@
   },
   "slither.solcPath": "",
   "slither.hiddenDetectors": [],
-  "solidity.compileUsingRemoteVersion": "v0.8.19+commit.7dd6d404"
+  "solidity.compileUsingRemoteVersion": "v0.8.19+commit.7dd6d404",
+  "solidity.defaultCompiler": "localFile"
```
Comment (GuillaumeNervoXS) [inline .vscode/settings.json:19]: ?

File: contracts/transmuter/configs/ProductionUSD.sol [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import "../libraries/LibOracle.sol";
+import { LibSetters } from "../libraries/LibSetters.sol";
+import { LibStorage as s } from "../libraries/LibStorage.sol";
+
+import { DummyDiamondImplementation } from "../../../scripts/generated/DummyDiamondImplementation.sol";
+
+import "../../utils/Constants.sol";
+import "../Storage.sol" as Storage;
+
+struct CollateralSetupProd {
```
Comment (GuillaumeNervoXS) [inline contracts/transmuter/configs/ProductionUSD.sol:17]: This can be put into a types .sol as shared

File: scripts/ConstantsArbitrum.s.sol [hunk 3]
```diff
@@ -12,3 +12,9 @@ address constant GUARDIAN = 0x55F01DDaE74b60e3c255BD2f619FEbdFce560a9C;
 address constant IMMUTABLE_CREATE2_FACTORY_ADDRESS = 0x0000000000FFe8B47B3e2130213B802212439497;
 address constant PROXY_ADMIN = 0x9a5b060Bd7b8f86c4C0D720a17367729670AfB19;
 address constant PROXY_ADMIN_GUARDIAN = 0xf2eDa0829E8A9CF53EBCB8AFCBb558D2eABCEF64;
+
```
Comment (GuillaumeNervoXS) [inline scripts/ConstantsArbitrum.s.sol:15]: We need to change those files with calls to sdk (you don't have to do it just for reference)

File: scripts/DeploySavingsNoImplem.s.sol [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(IMMUTABLE_CREATE2_FACTORY_ADDRESS, deployer, emptyData)
+        );
+        console.log("Proxy bytecode");
+        console.logBytes(initCode);
+        console.log("");
+
+        address computedAddress = create2Factory.findCreate2Address(salt, initCode);
+        console.log("Supposed to deploy: %s", computedAddress);
+        if (computedAddress != 0x004626A008B1aCdC4c74ab51644093b155e59A23) revert();
```
Comment (GuillaumeNervoXS) [inline scripts/DeploySavingsNoImplem.s.sol:59]: I guess you are aware but this is the stEUR address

File: scripts/DeployTransmuterWithoutFacets.s.sol [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+        // TODO change before actual deployment and replace with actual addresses inherited from other
+        // deployment
+
+        address agToken = address(new MockToken("agUSD", "agUSD", 18));
+        address treasury = address(new MockTreasury());
+        address accessControlManager = ACCESS_CONTROL_MANAGER;
+        /*
+        address agToken = AGEUR;
+        address treasury = 0x5d34839A3d4051f630D36e26698d53c58DD39072;
```
Comment (GuillaumeNervoXS) [inline scripts/DeployTransmuterWithoutFacets.s.sol:47]: Not the treasury but the dummy impl

File: .vscode/settings.json (hunk 1 above)
Comment (Picodes) [reply .vscode/settings.json:19]: +1, what does   "solidity.defaultCompiler": "localFile" means

File: scripts/ConstantsArbitrum.s.sol (hunk 3 above)
Comment (Picodes) [reply scripts/ConstantsArbitrum.s.sol:15]: +1, see in the governance contracts for the logic

File: scripts/Constants.s.sol [hunk 6]
```diff
@@ -14,11 +14,34 @@ address constant GUARDIAN = 0x0C2553e4B9dFA9f83b1A6D3EAB96c4bAaB42d430;
 address constant PROXY_ADMIN = 0x1D941EF0D3Bba4ad67DBfBCeE5262F4CEE53A32b;
 address constant PROXY_ADMIN_GUARDIAN = 0xD9F1A8e00b0EEbeDddd9aFEaB55019D55fcec017;
 address constant TREASURY_EUR = 0x8667DBEBf68B0BFa6Db54f550f41Be16c4067d60;
+address constant TREASURY_USD = 0xf8588520E760BB0b3bDD62Ecb25186A28b0830ee;
 address constant IMMUTABLE_CREATE2_FACTORY_ADDRESS = 0x0000000000FFe8B47B3e2130213B802212439497;
 address constant DEPLOYER = 0xfdA462548Ce04282f4B6D6619823a7C64Fdc0185;
-address constant TRANSMUTER = 0x00253582b2a3FE112feEC532221d9708c64cEFAb;
+address constant TRANSMUTER_EUR = 0x00253582b2a3FE112feEC532221d9708c64cEFAb;
```
Comment (GuillaumeNervoXS) [inline scripts/Constants.s.sol:20]: Should we add the TRANSMUTER_USD

File: scripts/ConstantsLinea.s.sol [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+pragma solidity ^0.8.19;
+
+import "contracts/utils/Constants.sol";
+
+address constant IMMUTABLE_CREATE2_FACTORY_ADDRESS = 0x0000000000FFe8B47B3e2130213B802212439497;
+address constant CHAIN_AGEUR = 0x1a7e4e63778B4f12a199C062f3eFdD288afCBce8;
+address constant PROXY_ADMIN = 0x1D941EF0D3Bba4ad67DBfBCeE5262F4CEE53A32b;
+address constant ACCESS_CONTROL_MANAGER = 0x4b1E2c2762667331Bc91648052F646d1b0d35984;
+
+address constant TREASURY_USD = 0x840b25c87B626a259CA5AC32124fA752F0230a72;
```
Comment (GuillaumeNervoXS) [inline scripts/ConstantsLinea.s.sol:11]: We didn't deploy the savings on Linea?

File: test/fuzz/SavingsTest.t.sol [hunk 8]
```diff
@@ -283,7 +284,9 @@ contract SavingsTest is Fixture, FunctionUtils {
         uint256 indexReceiver,
         uint256[2] memory elapseTimestamps
     ) public {
-        for (uint256 i; i < amounts.length; i++) amounts[i] = bound(amounts[i], 0, _maxAmount);
+        for (uint256 i; i < amounts.length; i++) {
```
Comment (GuillaumeNervoXS) [inline test/fuzz/SavingsTest.t.sol:287]: IS it the linter that adds those brackets?

File: test/fuzz/RedeemTest.t.sol [hunk 9]
```diff
@@ -403,8 +403,11 @@ contract RedeemTest is Fixture, FunctionUtils {
                 ) {
                     vm.expectRevert(bytes("SafeCast: value doesn't fit in 64 bits"));
                     shouldReturn = true;
-                } else if (amountBurnt > mintedStables) vm.expectRevert(Errors.TooBigAmountIn.selector);
-                else if (mintedStables == 0) vm.expectRevert(stdError.divisionError);
+                } else if (amountBurntBob > mintedStables) {
```
Comment (GuillaumeNervoXS) [inline test/fuzz/RedeemTest.t.sol:406]: No this should stay as `amountBurnt`

File: test/fuzz/RedeemTest.t.sol [hunk 10]
```diff
@@ -439,7 +442,16 @@ contract RedeemTest is Fixture, FunctionUtils {
                 );
                 assertApproxEqAbs(totalStable, mintedStables - amountBurnt, 3 wei);
             }
-            mintedStables = transmuter.getTotalIssued();
+
+            // Compute mintedStables while rounding up
+            uint128 normalizedStables = uint128(
```
Comment (GuillaumeNervoXS) [inline test/fuzz/RedeemTest.t.sol:447]: Was it breaking the test? 

Better to use the global system count than recomputing

Comment (0xtekgrinder) [reply test/fuzz/RedeemTest.t.sol:447]: It was breaking the test as getTotalIssued only returns the rounding down result but we need the rounding up result

File: test/fuzz/SavingsTest.t.sol (hunk 8 above)
Comment (0xtekgrinder) [reply test/fuzz/SavingsTest.t.sol:287]: It is me, revert back to without brackets

File: scripts/ConstantsLinea.s.sol (hunk 7 above)
Comment (sogipec) [reply scripts/ConstantsLinea.s.sol:11]: It wasn't deployed for agEUR prior

File: foundry.toml [hunk 11]
```diff
@@ -71,5 +74,5 @@ runs = 100
 
 [profile.ci.invariant]
 runs = 10
-depth = 30
+depth = 1
```
Comment (GuillaumeNervoXS) [inline foundry.toml:77]: to be rechanged

**Review by GuillaumeNervoXS — APPROVED** [review]

**Pull request merged**